tests/usm/usm_atomic_access.h

```diff
@@ -0,0 +1,173 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides common code for USM atomic access tests
+//
+*******************************************************************************/
+
+#ifndef __SYCL_CTS_TEST_USM_ATOMIC_ACCESS_H
+#define __SYCL_CTS_TEST_USM_ATOMIC_ACCESS_H
+
+#include "../../util/usm_helper.h"
+#include "../common/type_coverage.h"
+#include <chrono>
+#include <cstring>
+#include <limits.h>
+#include <thread>
+#include <type_traits>
+
+namespace usm_atomic_access {
+
+static auto get_scalar_types() {
+  static const auto scalar_types =
+      named_type_pack<short, unsigned short, int, unsigned int, long,
+                      unsigned long, float, double, long long,
+                      unsigned long long>({"short", "unsigned short", "int",
+                                           "unsigned int", "long",
+                                           "unsigned long", "float", "double",
+                                           "long long", "unsigned long long"});
+  return scalar_types;
+}
+
+/** @brief Return atomic aspect depending on the type of allocated memory
+ *  @retval AllocMemT USM allocation type
+ */
+template <sycl::usm::alloc AllocMemT>
+auto get_atomic_memory() {
+  if constexpr (AllocMemT == sycl::usm::alloc::host) {
+    return sycl::aspect::usm_atomic_host_allocations;
+  } else if constexpr (AllocMemT == sycl::usm::alloc::shared) {
+    return sycl::aspect::usm_atomic_shared_allocations;
+  } else {
+    static_assert(AllocMemT != AllocMemT,
+                  "This USM allocation type does not supported");
+  }
+}
+
+constexpr bool with_atomic64 = true;
+constexpr bool without_atomic64 = false;
+/** @brief Dummy functor that use in type coverage
+ *  @tparam CounterT Variable type from type coverage
+ *  @tparam UseAtomic64Flag std::integral_constant type
+ */
+template <typename CounterT>
+struct run_all_tests {
+  void operator()(sycl::queue &queue, sycl_cts::util::logger &log, bool use_atomic_flag,
+                  const std::string &typeName) {
+    if constexpr ((sizeof(CounterT) * CHAR_BIT) == 64) {
+      if (!use_atomic_flag) {
+        return;
+      }
+      if (!queue.get_device().has(sycl::aspect::atomic64)) {
+        log.note("Device does not perform 64-bit atomic operations");
+        return;
+      }
+    } else {
+      if (use_atomic_flag) {
+        return;
+      }
+    }
+    run_test_with_chosen_mem_type<CounterT>(queue, log, typeName);
```

Comment:
Actually I was thinking about keeping the `constexpr` approach you had in your first version but just in a more modern way
```suggestion
template <typename CounterT, bool use_atomic_flag>
struct run_all_tests {
  void operator()(sycl::queue &queue, sycl_cts::util::logger &log,
                  const std::string &typeName) {
    if constexpr ((sizeof(CounterT) * CHAR_BIT) == 64) {
      if constexpr (!use_atomic_flag) {
        return;
      }
      if (!queue.get_device().has(sycl::aspect::atomic64)) {
        log.note("Device does not perform 64-bit atomic operations");
        return;
      }
    } else {
      if constexpr (use_atomic_flag) {
        return;
      }
    }
    run_test_with_chosen_mem_type<CounterT>(queue, log, typeName);
```

Reply:
I can't provide `use_atomic_flag` as template parameter due to `for_all_types` (this function calls `run_all_tests`) does not support template parameters, that not equal to `typename` type

I suggest to not change `for_all_types` due to this function used across all project and this changes require do changes in many files and then verification that changed tests works correctly

Reply:
OK. As you wish.

Reply:
@keryell Thank You for understanding. Do You have any comments for this PR?